RP Cloud Overview and Security doc (#1504)
redpanda-data/documentation: docs/deploy/deployment-option/cloud/security/cloud-security-architecture/authorization/cloud-authorization.mdx

---
title: Authorization
---

<head>
    <meta name="title" content="Redpanda Cloud Authorization  | Redpanda Docs"/>
    <meta name="description" content="Learn how Redpanda Cloud uses IAM roles for authorization."/>
    <link rel="canonical" href="https://docs.redpanda.com/docs/deploy/deployment-option/cloud/security/cloud-security-architecture/authorization/" />
</head>

There are two types of authorization in Redpanda Cloud:

> - User authorizations

Reviewer: User authorization

>   User authorizations are managed by ACLs, and grant users permission to

Reviewer: Kafka ACLs

  perform specific types of operations on specific resources (such as topics,
  groups, clusters, or transactional IDs).

  For a detailed description of ACL terminology, commands, and guidance on
  how to manage users with ACLs, refer to [ACLs](../../../../../../../manage/security/authorization#acls).

> 

Reviewer: Yes, this wouldn’t be a blocker for this PR

>   User authorizations are managed by Kafka ACLs, and grant users permission to
>   perform specific types of operations on specific resources (such as topics,
>   groups, clusters, or transactional IDs).
> 
>   For a detailed description of ACL terminology, commands, and guidance on
>   how to manage users with ACLs, refer to [ACLs](../../../../../../../manage/security/authorization#acls).

Reviewer 2: ```suggestion
  User authorizations, managed by Kafka [ACLs](../../../../../../../manage/security/authorization#acls), grant users permission to
  perform specific types of operations on specific resources (such as topics,
  groups, clusters, or transactional IDs).
```

> - Agent authorizations

Reviewer: Agent authorization

>   how to manage users with ACLs, refer to [ACLs](../../../../../../../manage/security/authorization#acls).

Reviewer 2: suggest removing this line altogether, since there's a link to ACLs in the above sentence.

  When deploying an [agent](../../../../cloud-overview#agent) as part of BYOC cluster
  provisioning, Redpanda Cloud automatically assigns IAM policies to that agent.
  The IAM policy permissions granted to the agent provide it the authorization

>   required to enable Redpanda brokers to access AWS and GCP clusters.

Reviewer: ...authorization required to fully manage Redpanda Cloud clusters in AWS or GCP.

  :::important
  There are no explicit user actions associated with IAM policies. Nor do IAM
  policies apply or act as deployment permissions. Rather, IAM policy
  permissions apply to Redpanda Cloud agents _only_, and serve to provide Redpanda

>   agent access to AWS and GCP clusters so that Redpanda brokers can communicate

Reviewer: This paragraph sounds a bit weird. I'm not sure what exactly we are trying to highlight.

Author: I believe @bpraseed and I came up with this together. The goal was to convey that IAM policies only apply to agents upon deployment so that RP clusters can communicate with GCP and AWS--they do not act as permissions for the deployment. @bpraseed Can you weigh in on this one? Happy to remove as requested, but wanted to run by you first. Thx.

>   There are no explicit user actions associated with IAM policies. Nor do IAM
>   policies apply or act as deployment permissions. Rather, IAM policy
>   permissions apply to Redpanda Cloud agents _only_, and serve to provide Redpanda
>   agent access to AWS and GCP clusters so that Redpanda brokers can communicate

Reviewer 2: ```suggestion
  IAM policies do not apply or act as deployment permissions, and there are no explicit user actions associated with IAM policies. Rather, IAM policy
  permissions apply to Redpanda Cloud agents _only_, and serve to provide Redpanda
  agents access to AWS and GCP clusters so Redpanda brokers can communicate
```
@Feediver1 I'm not sure about this suggestion, but hoping to simplify the main point.

>   agents access to AWS and GCP clusters so Redpanda brokers can communicate

Reviewer: This sounds a bit confusing but I don’t want to block the PR more.

Author: Can you identify the specific source of confusion? This was the rewrite based on initial feedback.
